Shared/ThresholdKeyView.swift

```diff
@@ -239,6 +217,79 @@ struct ThresholdKeyView: View {
                         }
                     }
+                    
+                    HStack {
+                        Text("Enter SecurityQuestion password and reconstruct tkey & save share locally")
+                        Spacer()
+                        Button(action: {
+                            let alert = UIAlertController(title: "Enter Password", message: nil, preferredStyle: .alert)
+                            alert.addTextField { textField in
+                                textField.placeholder = "Password"
+                            }
+                            alert.addAction(UIAlertAction(title: "Cancel", style: .cancel))
+                            alert.addAction(UIAlertAction(title: "OK", style: .default, handler: { [weak alert] _ in
+                                guard let textField = alert?.textFields?.first, let answer = textField.text else { return }
+                                Task {
+                                    do {
+                                        
+                                        let result = try await SecurityQuestionModule.input_share(threshold_key: threshold_key, answer: answer)
+                                        if result {
+                                            // save this share locally
+                                            let shareIndexes = try threshold_key.get_shares_indexes()
+                                            
+                                            // let's get the device share index
+                                            var securityQuestionShareIndex = ""
+                                            if shareIndexes[0] == "1" {
+                                                securityQuestionShareIndex = shareIndexes[1]
+                                            } else {
+                                                securityQuestionShareIndex = shareIndexes[0]
+                                            }
+                                            
+                                            let share = try threshold_key.output_share(shareIndex: securityQuestionShareIndex, shareType: nil)
+                                            
+                                            guard let fetchKey = userData["publicAddress"] as? String else {
+                                                alertContent = "Failed to get public address from userinfo"
+                                                showAlert = true
+                                                return
+                                            }
+                                            
+                                            let saveId = fetchKey + ":0"
+                                            //save the security question share locally
+                                            try KeychainInterface.save(item: share, key: saveId)
```

Comment:
Shouldn't this share be saved separately instead of possibly overwriting an existing entry now or later?

Reply:
If you think about a normal application, I think it's more convenient for user experience if the process of saving a share happens seemingly without entering a password. Therefore, I don't think it's necessary to separate saving a share.

Reply:
The question is not about convenience, the question is about overwriting a previously saved share since the saveId would be the same, not so?

Reply:
fixed saveID